### DISCUSSION TITLE
Space usage v4 jtl_reporter

### DISCUSSION THREAD

**Discussion by fturibak** [ludeknovy/jtl-reporter#104] [Q&A]
Hi! :-)

I have maybe 9 long runs

![image](https://user-images.githubusercontent.com/59801436/144416408-e1f612ff-4782-4f10-ad28-4bbf032c144d.png)

that are using almost 30G of space!

It's like PG is storing all the data, not only aggregated ones...

`sudo du -sh data/`

`30G	data/`

**Comment by ludeknovy** [comment]
Hi @fturibak, 
see: https://github.com/ludeknovy/jtl-reporter/issues/33

**Reply by ludeknovy** [reply]
But it could be implemented as _delete samples after processing_ flag for scenario that would delete all the samples after full processing.

**Comment by fturibak** [comment]
Thanks for info!

I was wondering if you need samples once you've aggregated data. Maybe you can delete all unnecessary data once the aggregation completes..

 Anyhow  I'll keep this in mind

**Reply by ludeknovy** [reply]
Yes, that makes sense. In the past, I used them a couple of times in case of a bug. It was possible to fix the already existing data.

**Comment by fturibak** [comment]
Just from practical side. 

If I'm to run and publish results as part of CI/CD process every single day that's approximately 20-30GB of data per day = ~200GB per week = ~1TB per month

It's pretty expensive

**Reply by ludeknovy** [reply]
Agree. I will look into it soon.

**Reply by fturibak** [reply]
Thanks

**Reply by ludeknovy** [reply]
Try to update to the latest v4. Scenario settings now have `Delete sample data after processing` flag. Default is `false`, you need to enable it for given scenario in order to take effect. Also note, it will only work for newly uploaded samples.

**Reply by fturibak** [reply]
Could you also provide QUERY for deleting old data samples? In case someone else be needing to delete it..

**Reply by ludeknovy** [reply]
`DELETE FROM jtl.samples samples WHERE samples.item_id = $1;`

**Reply by fturibak** [reply]
Thanks!

**Reply by fturibak** [reply]
Hm, I've deleted records from `items` table.

`jtl_report=# SELECT pg_size_pretty( pg_total_relation_size('jtl.items')); `

` pg_size_pretty `

`----------------`
`
 48 kB`

`(1 row)`


But the size of database is still 28GB!

`jtl_report=# SELECT pg_size_pretty( pg_database_size('jtl_report'));`

` pg_size_pretty `

----------------
`
` 28 GB`
`
(1 row)
`

**Reply by fturibak** [reply]
After `VACUUM FULL `situation is better



jtl_report=# vacuum full;

VACUUM

jtl_report=# SELECT pg_size_pretty( pg_database_size('jtl_report'));

 pg_size_pretty 

----------------

 3444 MB

(1 row)

**Reply by ludeknovy** [reply]
Yes, `VACUUM FULL jtl.samples;` should make it.
